### PR TITLE
Update styled-system types to 3.0

### PR DESCRIPTION
**Pull request by LavaToaster**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jxnblk/styled-system/releases/tag/v3.0.0 + https://github.com/jxnblk/styled-system/releases/tag/v3.0.1 (This renames `css` to `mixed`)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

I'm opening this as a WIP because I've made enough changes that I want feedback asap, and I'm yet to actually use this in a project. (Sorting this package out was a precursor to using styled-system 😃in a new project of mine).

Also I'm aware of the implications of adding myself to the definitions header :)

### CODE REVIEW

**Comment by Lavoaster** [comment]
Ping @zephraph and @maoueh

File: types/styled-system/index.d.ts [hunk 1]
```diff
@@ -1,14 +1,60 @@
...
 //                 Ben McCormick <https://github.com/phobon>
 //                 Justin Bennett <https://github.com/zephraph>
 //                 Christopher Pappas <https://github.com/damassi>
 //                 Eloy Durán <https://github.com/alloy>
 //                 Matthieu Vachon <https://github.com/maoueh>
+//                 Adam Lavin <https://github.com/lavoaster>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.6
 
+export namespace util {
```
Comment (alloy) [inline types/styled-system/index.d.ts]: Using namespaces is discouraged. If this is another module in the package then you should add another ambient declaration file for it that follows the same FS layout as the package uses.

Comment (Lavoaster) [reply types/styled-system/index.d.ts]: Like it was before this commit? https://github.com/DefinitelyTyped/DefinitelyTyped/pull/27683/commits/a4f5821903a4f1d8f9dd9823a00361ca2bb297cd

The reasoning for that change is that `styled-system/utils` isn't a valid import, which why I brought it back into one file.

**Comment by Lavoaster** [comment]
Alright, I'm fairly happy with this so far (as I've started using it in a project). I'm gonna remove the `[WIP]` and say it's ready to merge once others are happy. 😄 

@alloy I left a comment on your review, let me know if it's acceptable. If it isn't then would you be able to provide an example on how to achieve what you want?

File: types/styled-system/index.d.ts (hunk 1 above)
Comment (alloy) [reply types/styled-system/index.d.ts]: Is it `styled-system/util` or something like `styled-system/dist/util`?

Comment (Lavoaster) [reply types/styled-system/index.d.ts]: The latter.

Comment (alloy) [reply types/styled-system/index.d.ts]: Ok, then that’s what the typings should follow 👍

Comment (Lavoaster) [reply types/styled-system/index.d.ts]: Alright

**Comment by alloy** [comment]
Awesome 👌

@Zephraph @maoueh Got any feedback on the actual type changes?

**Comment by maoueh** [comment]
+1 For me, nothing to say

**Comment by zephraph** [comment]
LGTM

**Comment by sheetalkamat** [comment]
@Lavoaster can you please refresh the PR by fixing merge conflicts. Thanks

**Comment by Lavoaster** [comment]
😢 That's a large merge conflict.

**Comment by Lavoaster** [comment]
@sheetalkamat Can you revert #27827 and merge this one instead? This achieves the same thing.

**Comment by Lavoaster** [comment]
@sheetalkamat Alright, I've resolved the merge conflict and fixed the additional lint errors that were raised.

**Comment by zephraph** [comment]
@Lavoaster thanks so much for you help! Awesome work.

**Pull request merged**